llama_stack/providers/remote/inference/vllm/vllm.py

```diff
@@ -148,14 +148,15 @@ async def _process_vllm_chat_completion_stream_response(
     async for chunk in stream:
         choice = chunk.choices[0]
         if choice.finish_reason:
+            args = tool_call_buf.arguments
             yield ChatCompletionResponseStreamChunk(
                 event=ChatCompletionResponseEvent(
                     event_type=event_type,
                     delta=ToolCallDelta(
                         tool_call=ToolCall(
                             call_id=tool_call_buf.call_id,
                             tool_name=tool_call_buf.tool_name,
-                            arguments=json.loads(tool_call_buf.arguments),
+                            arguments={} if not args else json.loads(args),
```

Comment:
what happens if the json is malformed? the `loads` is going to barf unexpectedly -- shouldn't that result in a final tool call which says "I couldn't parse it" vs. SSE stream just dying in the middle like what will happen in this (and existing) implementation?

Reply:
Great suggestion. Updated!